### PR TITLE
Improvement: Conditions serverside

### PR DESCRIPTION
**Pull request by SandrineBC**
# Summary
These are the server side changes, complementing the PR #1885 on the Bondage-Club client. 
It adds the new 'Conditions' to the server roundtrips and adds it to the mongoDB

# Testing
I tested this on my local installation and - as only existing server calls are used - all worked smoothly.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -811,9 +813,14 @@ function ChatRoomCharacterUpdate(data, socket) {
 				for (var A = 0; ((Acc.ChatRoom != null) && (A < Acc.ChatRoom.Account.length)); A++)
 					if ((Acc.ChatRoom.Account[A].ID == data.ID) && ChatRoomGetAllowItem(Acc, Acc.ChatRoom.Account[A]))
 						if ((typeof data.Appearance === "object") && Array.isArray(data.Appearance) && (data.Appearance.length >= 5) && (JSON.stringify(data.Appearance).length < 180000)) {
-							Database.collection("Accounts").updateOne({ AccountName : Acc.ChatRoom.Account[A].AccountName }, { $set: { Appearance: data.Appearance } }, function(err, res) { if (err) throw err; });
+							Database.collection("Accounts").updateOne(
```
Comment (elliesec) [inline app.js:816]: This doesn't appear to be updating the DB properly - logging out and back in loses the player's conditions. Have a look at the [MongoDB Node driver docs](http://mongodb.github.io/node-mongodb-native/3.6/api/Collection.html#updateOne) - it looks like you're passing in `{ $set: {Conditions: data.Conditions } }` as the `options` argument. You probably want something like this (I've not tested this yet):
```js
Database.collection("Accounts").updateOne(
    {AccountName: Acc.ChatRoom.Account[A].AccountName },
    { $set: { Appearance: data.Appearance, Conditions: data.Conditions } },
    function (err, res) { if (err) throw err; });
```

**Review by elliesec — CHANGES_REQUESTED** [review]

File: app.js (hunk 1 above)
Comment (elliesec) [reply app.js:816]: Actually, on further investigation, it doesn't seem to be this specifically, but _something_ is resetting the conditions on relog - I assume that's not intentional?

Comment (elliesec) [reply app.js:816]: Ignore me - I think I get it now.

**Comment by Ben987** [comment]
Clean-up of the GitHub pull requests.

**Pull request closed without merging**